src/sdk/accountspace_nb.js

```diff
@@ -181,28 +175,28 @@ class AccountSpaceNB {
         account_util._check_if_requesting_account_is_root_account(action, requesting_account, { });
         const is_truncated = false; // GAP - no pagination at this point
 
-        const root_name = requesting_account.name.unwrap();
-        // CORE CHANGES PENDING - START
+
         const requesting_account_iam_users = _.filter(system_store.data.accounts, function(acc) {
             if (!acc.name.unwrap().includes(IAM_SPLIT_CHARACTERS)) {
                 return false;
             }
-            return acc.name.unwrap().split(IAM_SPLIT_CHARACTERS)[1] === root_name;
+            return acc.name.unwrap().split(IAM_SPLIT_CHARACTERS)[1] === requesting_account._id.toString();
         });
```

Comment:
You're trying to get all the IAM users under the account, right?
You can filter by the `owner` filed that matches the requesting account.